pkg/model/tangle/transaction_storage.go

```diff
@@ -137,14 +137,24 @@ func (c *CachedTransaction) Release(force ...bool) {
 	c.metadata.Release(force...)
 }
 
-func transactionFactory(key []byte) (objectstorage.StorableObject, int, error) {
+func transactionFactory(key []byte, data []byte) (objectstorage.StorableObject, error) {
 	tx := hornet.NewTransaction(key[:49])
-	return tx, 49, nil
+
+	if err := tx.UnmarshalObjectStorageValue(data); err != nil {
+		return nil, err
+	}
+
+	return tx, nil
 }
 
-func metadataFactory(key []byte) (objectstorage.StorableObject, int, error) {
-	tx := hornet.NewTransactionMetadata(key[:49])
-	return tx, 49, nil
+func metadataFactory(key []byte, data []byte) (objectstorage.StorableObject, error) {
+	txMeta := hornet.NewTransactionMetadata(key[:49])
+
+	if err := txMeta.UnmarshalObjectStorageValue(data); err != nil {
```

Comment:
why keep the function? I thought OS will never call it now, but always use the factory

Reply:
Less changes in the code before we refactor everything.